compiler/backend/src/org/jetbrains/kotlin/codegen/inline/AnonymousObjectTransformer.kt

```diff
@@ -117,7 +117,8 @@ class AnonymousObjectTransformer(
         // When regenerating objects in lambdas, keep the old SMAP and don't remap the line numbers to save time.
         // The result is effectively the same anyway.
         val debugInfoToParse = debugInfo.takeIf { !inliningContext.isInliningLambda }
-        sourceMap = SMAPParser.parseOrCreateDefault(debugInfoToParse, sourceInfo, oldObjectType.internalName, 1, 65535)
+        val (firstLine, lastLine) = methodsToTransform.lineNumberRange()
```

Comment:
`constructor` is missed in `methodsToTransform`
Please add test if change wouldn't affect any test

Reply:
Fixed, thanks for pointing this out. I'm not surprised that no test broke: because objects inside inline functions always have SMAPs, hitting this bug would require having an init block in an object inside an inline lambda in a non-inline function while using the non-IR backend.